### PR TITLE
STAC-8752: Update the document for client certificate authentication

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: src/main/stackpack/resources/detailed-overview.md [hunk 1]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
 #### Enabling SAP check
 To enable the SAP check which collects the data from SAP host instance:
 
-Edit the `conf.yaml` file in your agent `conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+Edit the `conf.yaml` file in your agent `/etc/stackstate-agent/conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+
+To Connect to your SAP System, there are 2 ways to do so as explained below:-
```
Comment (TizianoPerrucci) [inline src/main/stackpack/resources/detailed-overview.md]: remove `-`

File: src/main/stackpack/resources/detailed-overview.md [hunk 2]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+      url: http://test-01   # <sap_host_url>   
+      user: test            # <username>
+      pass: test            # <password> 
+```
+**NOTE** - Make sure while using this mechanism, you have put `http` in the `url` of the config.
+
+
+##### 2. Client Certificate Authentication Mechanism
+
+This mechanism allow you to connect through the client certificate and private key for the certificate provided.
```
Comment (TizianoPerrucci) [inline src/main/stackpack/resources/detailed-overview.md]: This mechanism allow you to connect using the client certificate and the private key.

File: src/main/stackpack/resources/detailed-overview.md [hunk 3]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+      user: test            # <username>
+      pass: test            # <password> 
+```
+**NOTE** - Make sure while using this mechanism, you have put `http` in the `url` of the config.
+
+
+##### 2. Client Certificate Authentication Mechanism
+
+This mechanism allow you to connect through the client certificate and private key for the certificate provided. 
+The following new parameters are available :-
```
Comment (TizianoPerrucci) [inline src/main/stackpack/resources/detailed-overview.md]: remove `-`

File: src/main/stackpack/resources/detailed-overview.md [hunk 4]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+* `verify` - `True` or `False` depending on verifying the client certificate or not. By default, it's `False`.
```
Comment (TizianoPerrucci) [inline src/main/stackpack/resources/detailed-overview.md]: make sure this is `True` by default

File: src/main/stackpack/resources/detailed-overview.md [hunk 5]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+* `cert` - path containing the client side certificate
+* `keyfile` - path containing the private key for certificate
+
+_As an example, see the below config :-_
```
Comment (TizianoPerrucci) [inline src/main/stackpack/resources/detailed-overview.md]: remove `-`

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

**Review by TizianoPerrucci — APPROVED** [review]

File: src/main/stackpack/resources/detailed-overview.md [hunk 6]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
 To enable the SAP check which collects the data from SAP host instance:
 
-Edit the `conf.yaml` file in your agent `conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+Edit the `conf.yaml` file in your agent `/etc/stackstate-agent/conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+
+To Connect to your SAP System, there are 2 ways to do so as explained below:
+
+##### 1. HTTP Basic Authentication Mechanism
+
+This mechanism allow you to connect just using the `username` and `password` inside the `conf.yaml`.
```
Comment (kusztel) [inline src/main/stackpack/resources/detailed-overview.md]: This mechanism allow**s** you to connect using the `username`...

File: src/main/stackpack/resources/detailed-overview.md [hunk 7]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+      url: http://test-01   # <sap_host_url>   
+      user: test            # <username>
+      pass: test            # <password> 
+```
+**NOTE** - Make sure while using this mechanism, you have put `http` in the `url` of the config.
+
+
+##### 2. Client Certificate Authentication Mechanism
+
+This mechanism allow you to connect using the client certificate and the private key.
```
Comment (kusztel) [inline src/main/stackpack/resources/detailed-overview.md]: allow**s**

File: src/main/stackpack/resources/detailed-overview.md [hunk 8]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+      url: https://test-01      # <sap_host_url>   
+      user: test                # <username>
+      pass: test                # <password> 
+      verify: False
+      cert: /path/to/cert.pem   # <certificate_path>
+      keyfile: /path/to/key.pem # <keyfile_path>
 ```
+**NOTE** - Make sure while using this mechanism, you have put `https` in the `url` of the config.
+
+Once the configuration changes are done as explained above, restart the StackState Agent(s) using below command.
```
Comment (kusztel) [inline src/main/stackpack/resources/detailed-overview.md]: using the following command:

File: src/main/stackpack/resources/waitingfordata.md [hunk 9]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
 To enable the SAP check which collects the data from SAP host instance:
 
-Edit the `conf.yaml` file in your agent `conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+Edit the `conf.yaml` file in your agent `/etc/stackstate-agent/conf.d/sap.d` directory, replacing `<sap_host_name>` and `<sap_host_url>` with the unique name of SAP host to identify and URL to connect from your SAP instance.
+
+To Connect to your SAP System, there are 2 ways to do so as explained below:
+
+##### 1. HTTP Basic Authentication Mechanism
+
+This mechanism allow you to connect just using the `username` and `password` inside the `conf.yaml`.
```
Comment (kusztel) [inline src/main/stackpack/resources/waitingfordata.md]: allow**s** you to connect using...

File: src/main/stackpack/resources/waitingfordata.md [hunk 10]
```diff
@@ -9,20 +9,58 @@ The following prerequisites need to be met:
...
+      url: http://test-01   # <sap_host_url>   
+      user: test            # <username>
+      pass: test            # <password> 
+```
+**NOTE** - Make sure while using this mechanism, you have put `http` in the `url` of the config.
+
+
+##### 2. Client Certificate Authentication Mechanism
+
+This mechanism allow you to connect using the client certificate and the private key.
```
Comment (kusztel) [inline src/main/stackpack/resources/waitingfordata.md]: allow**s**

File: src/main/stackpack/resources/detailed-overview.md (hunk 6 above)
Comment (vishalxebia) [reply src/main/stackpack/resources/detailed-overview.md]: Done.

File: src/main/stackpack/resources/detailed-overview.md (hunk 7 above)
Comment (vishalxebia) [reply src/main/stackpack/resources/detailed-overview.md]: Done.

File: src/main/stackpack/resources/detailed-overview.md (hunk 8 above)
Comment (vishalxebia) [reply src/main/stackpack/resources/detailed-overview.md]: Done..

File: src/main/stackpack/resources/waitingfordata.md (hunk 9 above)
Comment (vishalxebia) [reply src/main/stackpack/resources/waitingfordata.md]: Done.

File: src/main/stackpack/resources/waitingfordata.md (hunk 10 above)
Comment (vishalxebia) [reply src/main/stackpack/resources/waitingfordata.md]: Done.

**Pull request merged**